### PR TITLE
Groovy & Spock Update and new test fixtures for Remote APIs

### PR DESCRIPTION
**Pull request by johnrengelman**
This change is [Reviewable](https://reviewable.io/reviews/ratpack/ratpack/1461)

### CODE REVIEW

File: ratpack-test/src/main/java/ratpack/test/mock/MockApi.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+ *     } >> groovyHandler {
+ *       render("remote ok")
+ *     }
+ *     resp.body.text == "remote ok"
+ *   }
+ * }
+ * }</pre>
+ * @since 1.7.0
+ */
+public class MockApi implements EmbeddedApp {
```
Comment (drmaas) [inline ratpack-test/src/main/java/ratpack/test/mock/MockApi.java]: This is cool - can you add documentation and/or convenience method to mock out a chain of requests? We often have multitple endpoints to mock together. Something like:

```
HandlerFactory factory = request -> Handlers.chain(request, chain -> chain.get()); 
MockApi mockApi = MockApi.of(request);
```

Maybe something like:
```
public static HandlerFactory of(Action<? super Chain> action) {
    return request -> Handlers.chain(request, action);
}
```

Comment (johnrengelman) [reply ratpack-test/src/main/java/ratpack/test/mock/MockApi.java]: well, the intent here is to create a test fixture that you can combine both spock mock invocation and end-to-end http requests, with this you can excercise a full AUT talking to embedded apps over real network, but easily configure how the embedded apps handle each request inline to a spec.

Its not intended as a replacement for wiremock as that does offer some lower level controls for testing things like connection resets and such, but as a mocking response behavior, this maybe be a little cleaner.

Comment (johnrengelman) [reply ratpack-test/src/main/java/ratpack/test/mock/MockApi.java]: I don't see how providing a chain would work, in that case why not just make an embedded app?

Comment (drmaas) [reply ratpack-test/src/main/java/ratpack/test/mock/MockApi.java]: @johnrengelman makes sense - this feature will functionally doing similar things that WireMock also does, via a different and more lightweight mechanism. Perhaps calling out when to use this vs. something heavier like WireMock or BetaMax would be useful to end users.

**Pull request merged**